specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json

```diff
@@ -0,0 +1,1285 @@
+          "x-ms-enum": {
+            "name": "configurationMode",
+            "modelAsString": true
+          },
+          "enum": [
+            "ApplyOnly",
+            "ApplyAndMonitor",
+            "ApplyAndAutoCorrect"
+          ]
+        },
+        "allowModuleOverwrite": {
+          "type": "string",
+          "description": "If true - new configurations downloaded from the pull service are allowed to overwrite the old ones on the target node. Otherwise, false",
+          "x-ms-enum": {
+            "name": "allowModuleOverwrite",
+            "modelAsString": true
+          },
+          "enum": [
+            "True",
+            "False"
+          ]
+        },
+        "actionAfterReboot": {
+          "type": "string",
+          "readOnly": false,
+          "description": "Specifies what happens after a reboot during the application of a configuration. The possible values are ContinueConfiguration and StopConfiguration",
+          "x-ms-enum": {
+            "name": "actionAfterReboot",
+            "modelAsString": true
+          },
+          "enum": [
+            "ContinueConfiguration",
+            "StopConfiguration"
+          ]
+        },
+        "refreshFrequencyMins": {
+          "type": "number",
+          "readOnly": false,
+          "default": 30,
+          "description": "The time interval, in minutes, at which the LCM checks a pull service to get updated configurations. This value is ignored if the LCM is not configured in pull mode. The default value is 30."
+        },
+        "rebootIfNeeded": {
+          "type": "string",
+          "readOnly": false,
+          "default": "False",
+          "description": "Set this to true to automatically reboot the node after a configuration that requires reboot is applied. Otherwise, you will have to manually reboot the node for any configuration that requires it. The default value is false. To use this setting when a reboot condition is enacted by something other than DSC (such as Windows Installer), combine this setting with the xPendingReboot module.",
+          "x-ms-enum": {
+            "name": "rebootIfNeeded",
+            "modelAsString": true
+          },
+          "enum": [
+            "True",
+            "False"
+          ]
+        },
+        "configurationModeFrequencyMins": {
+          "type": "number",
+          "readOnly": false,
+          "default": 15,
+          "description": "How often, in minutes, the current configuration is checked and applied. This property is ignored if the ConfigurationMode property is set to ApplyOnly. The default value is 15."
+        }
+      },
+      "description": "Configuration setting of LCM (Local Configuration Manager)."
+    },
+    "AssignmentReport": {
+      "properties": {
+        "id": {
```

Comment:
doesn't the "id" show up 3 times now?  The guestConfigAssignment has a "latestReportId" then this has a resource ID (which has the ID at the end of it) and a separate reportId property.  Why do we need all 3?

Reply:
i think there is some confusion  here :
"reportId": "7367cbb8-ae99-47d0-a33b-a283564d2cb1"
"id": "/subscriptions/mysubscriptionid/resourceGroups/myResourceGroupName/providers/Microsoft.Compute/virtualMachines/myvm/providers/Microsoft.GuestConfiguration/guestConfigurationAssignments/AuditSecureProtocol/reports/7367cbb8-ae99-47d0-a33b-a283564d2cb1",
targetResourceId == VM resourceID

only duplicate here is the latestReportId in the guestCOnfigurationAssignment resource at the top level and then the latestReport - would also have it. This is ok because we are not populating the latestReport object at all times right now. We are only sending it to ARG, api calls to control plane will have latestReport as null.